### PR TITLE
TG 71: add edit tasks

### PR DESCRIPTION
**Pull request by ilcosme**
This is my request for merging to master branch. I worked with the tasks resource, making the necessary functions to add a new task and also to edit an existing task. I worked with 2 main components (add/edit) and the corresponding forms to fill the asked data.

### CODE REVIEW

File: src/Components/Tasks/AddTask/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify(task)
+    });
+    const data = await res.json();
+
+    setTasks([...tasks, data]);
+
+    if (res.status === 201) {
+      console.log(data);
```
Comment (FefeTroanes) [inline src/Components/Tasks/AddTask/index.js:35]: Este `console.log` lo dejaste a proposito fer? Porque si no creo que podria eliminarse

File: src/Components/Tasks/EditTask/index.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  const editTask = async (task) => {
+    const res = await fetch(`http://localhost:8080/tasks/${taskID}`, {
+      method: 'PUT',
+      headers: {
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify(task)
+    });
+    const data = await res.json();
+    console.log('Data:', data);
```
Comment (FefeTroanes) [inline src/Components/Tasks/EditTask/index.js:38]: La misma consulta que antes

**Review by FefeTroanes — COMMENTED** [review]
Te deje unas constultas fer, es sobre unos `console.log` nada mas

File: src/Components/Tasks/AddTask/index.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      alert('Task added successfully');
+    } else if (res.status === 400 || res.status === 500) {
```
Comment (fernandocarbone1) [inline src/Components/Tasks/AddTask/index.js:37]: Sólo como comentario/sugerencia Fer: al tratarse de dos errores diferentes de status, tal vez el mensaje de alerta podría indicar si se trata de una mala petición o un problema del server (por ej:
return res.status(400).json({
message: error
});
Pero así como está, funciona igual. Por eso digo solo sugerencia.

**Review by fernandocarbone1 — APPROVED** [review]

File: src/Components/Tasks/AddTask/index.js (hunk 3 above)
Comment (ilcosme) [reply src/Components/Tasks/AddTask/index.js:39]: Intente pero en las controladoras no habia casos para status 400, asi que lo deje asi

File: src/Components/Tasks/EditTask/index.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    if (res.status === 200) {
+      console.log('Data:', data);
```
[the remark this answers is not in the document]
Comment (ilcosme) [reply src/Components/Tasks/EditTask/index.js]: Listo tirri

**Comment by FlorenciaEusebi** [comment]
## Test Results
Tested locally on alfonso-trackgenix-server.vercel.app

**Test Results: Failed** :warning:

Bugs found:
1. There are no buttons to add a task, edit a task or see existing tasks inside the Task Layout Section.

TASKS Layout

2. The input spaces inside the form should be centered.
TASKS Form

3. The BACK button (both in ADD and EDIT sections) is not working, it leads to https://alfonso-trackgenix-server.vercel.app/tasks
4.  Only existing Tasks can be added as new.
5. Edit section could not be tested, the tester was not able to edit existing tasks.

**Review by FlorenciaEusebi — CHANGES_REQUESTED** [review]
Please refer to https://github.com/BaSP-m2022/alfonso-trackgenix-app/pull/8#issuecomment-1146623427

File: public/index.html [hunk 5]
```diff
@@ -3,8 +3,12 @@
   <head>
     <meta charset="utf-8" />
 
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" 
+    integrity="sha512-KfkfwYDsLkIlwQp6LFnl8zNdLGxu9YAA1QvwINks4PhcElQSvqcyVLLD9aMhXd13uQjoXtEKNosOWaZqXgel0g==" 
+    crossorigin="anonymous" referrerpolicy="no-referrer" />
     <link rel="icon" type="image/png" sizes="32x32" href="%PUBLIC_URL%/favicon-32x32.png">
     <link rel="icon" type="image/png" sizes="16x16" href="%PUBLIC_URL%/favicon-16x16.png">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" integrity="sha512-KfkfwYDsLkIlwQp6LFnl8zNdLGxu9YAA1QvwINks4PhcElQSvqcyVLLD9aMhXd13uQjoXtEKNosOWaZqXgel0g==" crossorigin="anonymous" referrerpolicy="no-referrer" />
```
Comment (martinalbrnz) [inline public/index.html:11]: Ojo que ya esta el link

File: public/index.html [hunk 6]
```diff
@@ -14,6 +18,7 @@
       content="Web site created using create-react-app"
     />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" integrity="sha512-KfkfwYDsLkIlwQp6LFnl8zNdLGxu9YAA1QvwINks4PhcElQSvqcyVLLD9aMhXd13uQjoXtEKNosOWaZqXgel0g==" crossorigin="anonymous" referrerpolicy="no-referrer" />
```
Comment (martinalbrnz) [inline public/index.html:21]: Ojo que ya esta el link x2

File: src/Components/Tasks/List/Row/index.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+          ? assignedEmployee.map((employee) => employee.firstName)
+          : 'no tiene id'}
+      </td>
+      <td>{new Date(startDate).toLocaleDateString()}</td>
+      <td>{status}</td>
+      <td>
+        <i className="fa-solid fa-xmark" onClick={() => props.deleteTask(id)}></i>
+      </td>
+      <td>
+        <a href={`/tasks-edit/${id}`}>
```
Comment (martinalbrnz) [inline src/Components/Tasks/List/Row/index.jsx:32]: try with 
`
<a href={`/tasks-edit?id=${id}`}>
`

**Review by FlorenciaEusebi — APPROVED** [review]

**Pull request merged**